drivers/hmis/app/graphql/types/hmis_schema/project.rb

```diff
@@ -178,6 +182,8 @@ def unit_types
 
     # TODO use dataloader
     def units(**args)
+      raise HmisErrors::ApiError, 'Access denied' unless current_permission?(entity: object, permission: :can_view_units)
```

Comment:
Let's discuss whether to raise or just return empty?
- This is an attribute on the project object, so we don't want to raise and prevent someone from returning other project attrs that they do have permission to see
- However, from the frontend schema right now it is only used in purpose-built queries for the units page, so I figured that raising was the correct behavior

Reply:
I think return empty. That's our default for "field level authorization", and this feels similar. Also think about the "viewable_by" scopes which we use when resolving most fields like this. (In that case if the user has no view access, it's just empty).
I can imagine a future page where maybe the Project Overview page has a summary of unit availability stats. We would want to be able to just make a single query for that may include units, and not worry about that throwing an error if the user lacks access to view it.